tests/day1-tests/test_expense_report_fix.rb

```diff
@@ -0,0 +1,112 @@
+#: Unit tests for the Day-1 solutions.
+
+require 'test/unit'
+require_relative '../common'
+require_relative SRC_DIR + 'day1/expense_report_fix'
+
+# Isolate tests in a module to avoid constants' name collisions/side effects
+module TestExpenseReportWrapper
+
+  MY_DIR = File.expand_path File.dirname(__FILE__)
+  REF_INPUT_FILE = MY_DIR + '/ref-input-day1.txt'
+  PUZZLE_INPUT_FILE = SRC_DIR + 'day1/puzzle-input-day1.txt'
+
+  REFERENCE_INPUT = AocUtils.to_int_array(REF_INPUT_FILE)
+  PUZZLE_INPUT = AocUtils.to_int_array(PUZZLE_INPUT_FILE)
+
+  class TestArrayExtentions < Test::Unit::TestCase
+
+    def test_scan_pairs__size_3
+      res = []
+      ExpenseReportFix.scan_pair_combinations(3) { |i, j| res << [i, j] }
+      assert_equal [[0, 1], [0, 2], [1, 2]], res
+    end
+
+    def test_scan_pairs__size_0
+      res = []
+      ExpenseReportFix.scan_pair_combinations(0) { |i, j| res << [i, j] }
+      assert_equal [], res
+    end
+
+    def test_scan_pairs__size_1
+      res = []
+      ExpenseReportFix.scan_pair_combinations(1) { |i, j| res << [i, j] }
+      assert_equal [], res
+    end
+
+    def test_scan_triads__size_4
+      res = []
+      ExpenseReportFix.scan_triad_combinations(4) { |i, j, k| res << [i, j, k] }
```

Comment:
You can fix this need to pass a block by doing this: https://stackoverflow.com/a/66059990/135665

Reply:
Yep, nice, thank you! Done in e07ebc5.